app/src/main/java/com/gh4a/utils/rx/RxTools.java

```diff
@@ -0,0 +1,102 @@
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import com.gh4a.BasePagerActivity;
+import com.gh4a.R;
+import com.gh4a.utils.UiUtils;
+import com.philosophicalhacker.lib.RxLoader;
+
+public class RxTools {
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+            return observable
+                    .compose(handleError(activity))
+                    .compose(rxLoader.makeObservableTransformer(id));
+        };
+    }
+
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id, boolean refresh) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+
+            if(refresh) {
+                return observable
+                        .compose(handleError(activity))
+                        .compose(rxLoader.makeObservableTransformer(id, refresh));
+            } else return observable.compose(handle(activity, id));
+        };
+    }
+
+    // Error handler showing SnackBar
+    public static <T> ObservableTransformer<T, T> onErrorSnackbar(Activity activity, int id, View rootLayout, String errorMessage, int messageRes) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+            return observable
+                    .compose(rxLoader.makeObservableTransformer(id))
+                    .doOnSubscribe(disposable -> showProgressDialog(activity, messageRes))
+                    .doOnTerminate(() -> dismissProgressDialog())
+                    .doOnError(error -> {
+                        Snackbar.make(rootLayout, errorMessage, Snackbar.LENGTH_LONG)
+                                .setAction(R.string.retry, (View.OnClickListener) activity)
+                                .show();
+                    });
+        };
+    }
+
+    public static <T> ObservableTransformer<T, T> handleError(Activity activity) {
+        return observable -> observable
+                .doOnError(error -> {
+                    if (activity instanceof BasePagerActivity) {
+                        BasePagerActivity act = (BasePagerActivity) activity;
+                        act.setErrorViewVisibility(true);
+                    } else {
+                        // other cases
+                    }
+                });
+    }
+
+    public static <T> ObservableTransformer<T, T> applySchedulers() {
+        return observable -> ((Observable)observable).subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread());
+    }
+
+
+    public static ProgressDialogFragment mProgressDialogFragment;
```

Comment:
As mentioned before: please don't keep static references to UI stuff. The fragment instance reference needs to be kept in the call chain of the transformer creation or in the transformer instance.